scheduler/src/cook/compute_cluster.clj

```diff
@@ -50,6 +50,12 @@
   (restore-offers [this pool-name offers]
     "Called when offers are not processed to ensure they're still available.")
 
+  (get-outstanding-synthetic-pods [this pool-name]
+    "Returns a list of outstanding synthetic pods in the pool.")
+
+  (set-synthetic-pods-counters [this pool-name num-synthetic-pods]
```

Comment:
Is there something forcing this into the API? Can't the implementation be outside of the compute cluster; you can get the compute cluster name with `get-compute-cluster` and incorporate that.

Reply:
I tried that, but because the autoscaling code in the scheduler uses the compute cluster API and doesn't actually know if the compute cluster is Kubernetes or Mesos, separating this function out would break the abstraction.

Reply:
Aaah. Yes, I see. This code needs `max-total-synthetic-pods` which comes from `scheduler-config` which is not visible outside. (And thats worth a comment)

Then new proposal, why fetch total synthetic pods and then immediately feed it back? Why not replace both functions with something like 
`(log-synthetic-pod-metrics [this pool-name])` thats a noop for k8s?

Reply:
I did it this way because we already need to calculate the outstanding synthetic pods for the `autoscale!` function, and I didn't want to replicate the calculation in the case when we do autoscale. If you think repeating that work is fine, I can update it to be a self-contained function that does all its calculations like you suggest.